enterprise/server/backends/redis_cache/redis_cache.go

```diff
@@ -37,16 +43,34 @@ type Cache struct {
 	cutoffSizeBytes int64
 }
 
-func NewCache(redisClient redis.UniversalClient, maxValueSizeBytes int64) *Cache {
-	c := &Cache{
+func Register(env environment.Env) error {
+	crcc := redis_client.CacheRedisClientConfig()
+	if crcc == nil {
+		return nil
+	}
+	if _, ok := env.GetCache().(*composable_cache.ComposableCache); ok {
+		// Cache has already been composed, don't do it again.
```

Comment:
nit: same as above, can we also log a warning here so that people can fix their flags?